lib/saluki-config/src/refresher.rs

```diff
@@ -0,0 +1,89 @@
+use std::sync::Arc;
+
+use crate::GenericConfiguration;
+use reqwest;
+use saluki_error::GenericError;
+use serde::Deserialize;
+use tokio::time::{sleep, Duration};
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct ConfigResponse {
+    api_key: String,
+}
+
+/// Configuration for setting up `ConfigRefresher`.
+#[derive(Default, Deserialize)]
+pub struct ConfigRefresherConfiguration {
+    api_key: String,
+    auth_token_file_path: String,
+}
+
+/// The most recent configuration retrieved from the datadog-agent.
+#[derive(Default)]
+pub struct ConfigRefresher {
+    token: String,
+    api_key: String,
+}
+
+#[allow(unused)]
+impl ConfigRefresherConfiguration {
+    /// Creates a new `ConfigRefresherConfiguration` from the given configuration.
+    pub fn from_configuration(config: &GenericConfiguration) -> Result<Self, GenericError> {
+        Ok(config.as_typed()?)
+    }
+    /// Create `ConfigRefresher` from `ConfigRefresherConfiguration`
+    pub fn build(&self) -> Result<ConfigRefresher, GenericError> {
+        let raw_bearer_token = std::fs::read_to_string(&self.auth_token_file_path)?;
+        Ok(ConfigRefresher {
+            token: raw_bearer_token,
+            api_key: self.api_key.clone(),
+        })
+    }
+}
+impl ConfigRefresher {
+    /// Start a task that queries the datadog-agent config endpoint every 15 seconds.
+    pub fn spawn_refresh_task(self: Arc<Self>) {
+        tokio::spawn(async move {
+            loop {
+                match self.query_agent().await {
+                    Ok(_) => {
+                        // todo()!
+                    }
+                    Err(_e) => {
+                        // todo()!
+                    }
+                }
+                sleep(Duration::from_secs(15)).await; // Wait for 15 seconds
+            }
+        });
+    }
+    /// Query the datadog-agent config endpoint for the latest config
+    #[allow(unused)]
+    pub async fn query_agent(&self) -> Result<(), GenericError> {
+        let url = format!("https://localhost:5004/config/v1/");
+        let client = reqwest::ClientBuilder::new()
```

Comment:
Using `reqwest` here since I needed `.danger_accept_invalid_certs` for the request to work.

The curl command i was trying to imitate was 
` curl  -H "authorization: Bearer $(cat /etc/datadog-agent/auth/auth_token)" -k https://localhost:5004/config/v1/`